src/main/scala/beam/router/Modes.scala

```diff
@@ -245,3 +247,69 @@ object Modes {
   }
 
 }
+
+object TourModes {
+  import beam.router.Modes.BeamMode
+  import beam.router.Modes.BeamMode._
+
+  sealed abstract class BeamTourMode(
+    val value: String,
+    val vehicleCategory: VehicleCategory,
+    val allowedBeamModes: Seq[BeamMode],
+    val allowedBeamModesForFirstAndLastLeg: Seq[BeamMode]
+  ) extends StringEnumEntry {
+
+    import BeamTourMode._
+
+    def isVehicleBased: Boolean = this match {
+      case WALK_BASED => false
+      case _          => true
+    }
+  }
+
+  object BeamTourMode extends StringEnum[BeamTourMode] with StringCirceEnum[BeamTourMode] {
+
+    override val values: immutable.IndexedSeq[BeamTourMode] = findValues
+
+    val enabledModes: Map[BeamMode, Seq[BeamMode]] =
+      Map[BeamMode, Seq[BeamMode]](CAR -> Seq(DRIVE_TRANSIT), BIKE -> Seq(BIKE_TRANSIT))
+
+    // TODO: Also allow use of shared bikes/cars in walk based tours
+    case object WALK_BASED
+        extends BeamTourMode(
```

Comment:
Think we could have 
`id: Id[BeamVehicleType]`
And then distinguish shared bike by
`def isSharedVehicle: Boolean = id.toString.startsWith("sharedVehicle")`
And then filter shared bike by
```
val bikeVehicleType = beamScenario.vehicleTypes.values
  .filterNot(_.isSharedVehicle)
  .find(_.vehicleCategory == VehicleCategory.Bike)
  .getOrElse(throw new RuntimeException("Bike not found in vehicle types."))
```
And then create a new object of shared bike to put in

Reply:
I think you're on to something -- how about rather than giving this an `id` attribute we change the `allowedBeamModes` to a private attribute the class a new method like `def allowedBeamModes(vehicles: Vector[VehicleOrToken]): Seq[BeamMode]` that checks whether the vehicles are shared or not before returning the allowed beam modes, having `BIKE_BASED` not return `BIKE` if the only bikes available are shared bikes.